<launch>
  <arg name="voxel_size" default="0.2"/>
  <arg name="voxels_per_side" default="16" />
  <arg name="truncation_distance" default="0.6" />
  <arg name="ray_length" default="5" />

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />
  <arg name="world_frame" default="world_origin" />

  <group ns="$(arg UAV_NAME)">
    <node pkg="motion_planning" type="KinoNBVReal" name="planner_node" output="screen">
      <rosparam file="$(find motion_planning)/config/KinoNBVplannerReal.yaml" />
      <param name="uav_namespace" value="$(arg UAV_NAME)" />
      <!-- Voxblox Planner -->
      <remap from="~esdf_map_in" to="voxblox_node/esdf_map_out" />
      <remap from="~tsdf_map_in" to="voxblox_node/tsdf_map_out" />

      <param name="use_tf_transforms" value="false" />
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="publish_traversable" value="false" />

      <!-- Publishers -->
      <remap from="~visualization_marker_out" to="visualization_marker_out" />
      <remap from="~initial_reference_out" to="mavros/setpoint_position/global" />
      <remap from="~reference_out" to="reference_out" />
      <remap from="~frustum_out" to="frustum_out" />
      <remap from="~unknown_voxels_out" to="unknown_voxels_out" />
      <remap from="~simulation_ready" to="simulation_ready" />
      <!-- Subscribers -->
      <remap from="~global_state_in" to="mavros/global_position/global" />
      <remap from="~local_pose_in" to="mavros/local_position/pose" />
      <remap from="~local_velocity_in" to="mavros/local_position/velocity_local" />
      <remap from="~control_manager_diag_in" to="control_manager/diagnostics" />
      <!-- Services clients-->
      <remap from="~trajectory_reference_out" to="control_manager/trajectory_reference" />
      <!-- Services servers-->
      <remap from="~start_in" to="~start" />
      <remap from="~stop_in" to="~stop" />

    </node>

  </group>

</launch>
